source/common/http/http1/codec_impl.cc

```diff
@@ -67,7 +70,7 @@ void StreamEncoderImpl::encode100ContinueHeaders(const HeaderMap& headers) {
 void StreamEncoderImpl::encodeHeaders(const HeaderMap& headers, bool end_stream) {
   bool saw_content_length = false;
   headers.iterate(
-      [](const HeaderEntry& header, void* context) -> HeaderMap::Iterate {
+      [this](const HeaderEntry& header, void* context) -> HeaderMap::Iterate {
```

Comment:
FWIW, this function used a C-style callback originally for performance reasons when I was optimizing for the horse race case. I would leave a comment here about looking at perf, or potentially just pre-create the lambda at construction time so that in the keep alive case we don't need any extra allocations.

Reply:
Do you think it's worth having multiple versions of iterate, one with C-style callback and one with std::function? afaik std::function adds some overhead compared to C-style callbacks due to the additional indirection

Reply:
Turns out I was able to do this without changing the signature, since I could just use the encoder that was already being passed as the context to resolve the formatter.